File: code-examples/pymongo/authentication/sha256.py
from pymongo import MongoClient
from urllib.parse import quote_plus

# Replace the following with values for your environment.
username = quote_plus("<username>")
password = quote_plus("<password>")
cluster_url = "<MongoDB cluster url>"

auth_mechanism = "SCRAM-SHA-256"

# Replace the following with your MongoDB deployment's connection string.
uri = f"mongodb+srv://{username}:{password}@{cluster_url}/?authMechanism={auth_mechanism}"

# Create a new MongoClient
client = MongoClient(uri)

def run():
    try:
        # Establish and verify connection
        client.admin.command('ping')
        print("Connected successfully to server")
    except Exception as e:
        print(e)
    finally:
        # Ensures that the client will close when you finish/error
        client.close()

run()

File: code-examples/java-rs/authentication/sha256.java
import com.mongodb.ConnectionString;
import com.mongodb.reactivestreams.client.MongoClient;
import com.mongodb.reactivestreams.client.MongoClients;
import com.mongodb.reactivestreams.client.MongoDatabase;
import org.bson.Document;

import java.util.concurrent.CountDownLatch;
import java.util.concurrent.TimeUnit;

public class Main {
    public static void main(String[] args) {
        String username = "<username>";
        String password = "<password>";
        String clusterUrl = "<MongoDB cluster url>";
        String authMechanism = "SCRAM-SHA-256";

        String uri = "mongodb+srv://" + username + ":" + password + "@" + clusterUrl + "/?authMechanism=" + authMechanism;

        ConnectionString connectString = new ConnectionString(uri);

        MongoClient mongoClient = MongoClients.create(connectString);

        MongoDatabase database = mongoClient.getDatabase("admin");

        CountDownLatch latch = new CountDownLatch(1);

        database.runCommand(new Document("ping", 1)).subscribe(new OperationSubscriber<Document>(){
            @Override
            public void onNext(Document document) {
                System.out.println("Connected successfully to server");
            }

            @Override
            public void onError(Throwable throwable) {
                System.out.println("Error: " + throwable.getMessage());
                latch.countDown();
            }

            @Override
            public void onComplete() {
                System.out.println("Command completed successfully");
                latch.countDown();
            }
        });

        try {
            if (!latch.await(30, TimeUnit.SECONDS)) {
                System.out.println("Operation took too long");
            }
        } catch (InterruptedException e) {
            e.printStackTrace();
        } finally {
            mongoClient.close();
        }
    }
}

abstract class OperationSubscriber<T> implements Subscriber<T> {
    @Override
    public void onSubscribe(Subscription subscription) {
        subscription.request(1);
    }

    @Override
    public abstract void onNext(T t);

    @Override
    public abstract void onError(Throwable throwable);

    @Override
    public abstract void onComplete();
}
